### PR TITLE
[OPENMIND-14] Implement repositories for all the models

### PR DESCRIPTION
**Pull request by pikulo-kama**

### CODE REVIEW

File: src/main/java/com/arthurdrabazha/openmind/controller/CategoryController.java [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+
+    @Autowired
+    public CategoryController(CategoryRepository categoryService) {
+        this.categoryService = categoryService;
+    }
+
+    public String index() {
+        categoryService.
+    }
+}
```
Comment (MarrMika) [inline src/main/java/com/arthurdrabazha/openmind/controller/CategoryController.java:18]: I see the mess in this class. Delete this one, please.

File: src/main/java/com/arthurdrabazha/openmind/repository/CategoryRepository.java [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+
+import com.arthurdrabazha.openmind.model.Category;
+import com.arthurdrabazha.openmind.model.Post;
+import com.arthurdrabazha.openmind.model.User;
+import org.springframework.data.jpa.repository.JpaRepository;
+import org.springframework.stereotype.Repository;
+
+import java.util.List;
+
+@Repository
```
Comment (MarrMika) [inline src/main/java/com/arthurdrabazha/openmind/repository/CategoryRepository.java:11]: It is not necessary to use @Repository annotation here

File: src/main/java/com/arthurdrabazha/openmind/repository/PostRepository.java [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+    List<Post> findAllByAuthor(User author);
+
+    List<Post> findAllByCategoriesContains(Category category);
+
+    List<Post> findAllByTopicContains(String topic);
+
+    List<Post> findPostsByDislikesNotNullOrderByLikesAsc();
+
+    List<Post> findPostsByDislikesNotNullOrderByLikesDesc();
+
```
Comment (MarrMika) [inline src/main/java/com/arthurdrabazha/openmind/repository/PostRepository.java:21]: Have you checked whether all these methods work correctly?

**Review by MarrMika — CHANGES_REQUESTED** [review]
In general good, but we need improvement. Please, fix comments!

File: src/main/java/com/arthurdrabazha/openmind/controller/CategoryController.java (hunk 1 above)
Comment (pikulo-kama) [reply src/main/java/com/arthurdrabazha/openmind/controller/CategoryController.java]: Sorry, my mistake. I will delete this one.

File: src/main/java/com/arthurdrabazha/openmind/repository/CategoryRepository.java (hunk 2 above)
Comment (pikulo-kama) [reply src/main/java/com/arthurdrabazha/openmind/repository/CategoryRepository.java:11]: But in that case I'll not be able to use repository as bean for injection in my services. Isn't it?

Comment (MarrMika) [reply src/main/java/com/arthurdrabazha/openmind/repository/CategoryRepository.java]: Nope, you will be able to. It's a specification, not an implementation

File: src/main/java/com/arthurdrabazha/openmind/repository/CategoryRepository.java [hunk 4]
```diff
@@ -0,0 +1,17 @@
+package com.arthurdrabazha.openmind.repository;
+
+import com.arthurdrabazha.openmind.model.Category;
+import org.springframework.data.jpa.repository.JpaRepository;
+import org.springframework.data.jpa.repository.Query;
+
+import java.util.List;
+
+public interface CategoryRepository extends JpaRepository<Category, Short> {
+
```
Comment (MarrMika) [inline src/main/java/com/arthurdrabazha/openmind/repository/CategoryRepository.java:10]: Why Category PK has a Short type?

**Review by MarrMika — APPROVED** [review]

File: src/main/java/com/arthurdrabazha/openmind/repository/CategoryRepository.java (hunk 4 above)
Comment (pikulo-kama) [reply src/main/java/com/arthurdrabazha/openmind/repository/CategoryRepository.java:10]: I think, that will be better, to use type which consumes less space. Anyway, I don't think that count of categories will ever exceed max value of Short.

Comment (MarrMika) [reply src/main/java/com/arthurdrabazha/openmind/repository/CategoryRepository.java:10]: Please, don't mix between Long and Short. By default, bigint type in PostgreSQL equals 64 bits that is the most appropriate to use Long

Comment (pikulo-kama) [reply src/main/java/com/arthurdrabazha/openmind/repository/CategoryRepository.java:10]: Okay, I'll keep that in mind. I will change PK type to Long, and then merge thia branch with master

**Pull request merged**